src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -242,4 +276,12 @@ Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         return value;
     }
   }
+
+  public enum BehaviorOnNullValues {
+    IGNORE,
+    DELETE,
+    FAIL;
+
+    public static final BehaviorOnNullValues DEFAULT = IGNORE;
+  }
```

Comment:
Not opposed to overriding `toString()` but after following your suggestion below about making the `behavior.on.null.values` property case-insensitive, is there still any reason to?

Reply:
In `enum`s like this one, you almost always want to override `toString`. If you don't you'll get by default `name()` which is the name of the enum member in upper case. 

But we are going to use lower case, so we want a way to print `BehaviorOnNullValues.IGNORE` as `ignore`. That's regardless of the suggestion in config. 

Now, depending what `define` expects in `ConfigDef` you should use `BehaviorOnNullValues.DEFAULT.toString()` or `BehaviorOnNullValues.DEFAULT` instead of hardcoding `"ignore"` at least in the default value for define. Similar elsewhere. Hopefully, we won't have to hardcode a value in any place.

Reply:
> But we are going to use lower case, so we want a way to print `BehaviorOnNullValues.IGNORE` as `ignore`.
Right, so the question I had was that if we end up making the field case-insensitive (so that `ignore`, `Ignore`, `IGNORE`, and even `iGnOrE`) are all accepted, does it actually matter that the enums appear as all lowercase when referenced to the user?

Reply:
yes, we should be a bit flexible w.r.t what we accept, but specific to our suggestion (and consistent to what we print in log messages). If we advice users to give lower case values in our docs (and I agree with your initial suggestion), that's what we should print also. 

Again, this is very common for such enums.

Reply:
Have a look at [ConnectorType](https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorType.java) for an example of adding a static lookup method to the enum to handle the case-insensitive lookup. But @kkonstantine is right that toString and docs should use lowercase.

Reply:
Alright, sounds good. Will adapt the `forValue()` method from the ConnectorType enum to enable case-insensitive reading (as long as all the enum values are uppercase), and override `toString()` to return `name().toLowerCase()`.